### DISCUSSION TITLE
About SSO to shellhub website

### DISCUSSION THREAD

**Discussion by bonnlee2024** [shellhub-io/shellhub#4044] [Q&A]
I want to use a third-party system to log in to shellhub web site by SSO, without entering the user name and password. How can I achieve this?

I currently have a business system that needs to integrate with ShellHub, and I want to provide users with a better experience by automatically authenticating and logging in to ShellHub through a link without requiring them to enter their account and password information again.

My idea is to first configure the account in ShellHub and store it in my business system, and then verify it through openApi. However, I am not sure how to do this, so please guide me and thank you.

**Comment by heiytor** [comment]
Hi @bonnlee2024,

ShellHub primarily supports authentication via API keys and username/password. API keys are tied to specific namespaces and can be used for all endpoints (except for the user-related ones), enabling you to create an interface where users won't need to log in repeatedly. However, username/password authentication is the only method for accessing the UI, with the login session being valid for 72 hours.

If you want to redirect your users to ShellHub's UI without requiring them to reauthenticate, there are two potential approaches:

1. **Store the Username and Password**: You could store the user's credentials, authenticate them, and save the token in ShellHub's local storage. This method, however, poses security risks as storing passwords is generally unsafe, and accessing another site's local storage might be problematic.
  
2. **Extend the Token Duration**: If you're using a community instance, you might consider extending the token's duration. While this would keep users logged in for longer, it increases security risks since the token cannot be invalidated if it gets leaked.

Let me know if you need further assistance.
